.github/CODEOWNERS

```
@@ -3,15 +3,146 @@
 
 *                                    @MetaMask/engineering
```

We should add protection for the CODEOWNERS file. Currently (and hypothetically), anyone in `@MetaMask/engineering` could unilaterally alter or remove this ruleset that we've painstakingly put together.

```suggestion
*                                    @MetaMask/engineering
/.github/                            @MetaMask/wallet-framework-engineers 
```

> To protect a repository fully against unauthorized changes, you also need to define an owner for the CODEOWNERS file itself. The most secure method is to define a CODEOWNERS file in the .github directory of the repository and define the repository owner as the owner of either the CODEOWNERS file (/.github/CODEOWNERS @owner_username) or the whole directory (/.github/ @owner_username).

> -- https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-and-branch-protection